pkg/generator/generate_inputs_test.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package generator_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/vmware-tanzu/carvel-secretgen-controller/pkg/generator"
+)
+
+func TestAddSucceedsWithEmptyAnnotations(t *testing.T) {
+	err := generator.GenerateInputs{}.Add(nil)
+	assert.Equal(t, nil, err)
```

Comment:
here you're testing that there's no assertions,  but i think that the map you created on line 22 (https://github.com/vmware-tanzu/carvel-secretgen-controller/pull/92/files#diff-c07fd47e85b29b850abf835f11f15dadd5a05dee6387a5aaa4883e2eabf63960R22) goes out of scope when the function returns.  That was why we pointed you towards the place that calls the function as likely the right place to create the map. 

  Can you change this test to assert that if it's called with a Secret with no annotations, then the annotation we added in that function is actually present on the secret afterwards?

Reply:
Yep, [my bad](https://dave.cheney.net/2017/04/29/there-is-no-pass-by-reference-in-go#:~:text=But%20maps%20and%20channels%20are%20references%2C%20right%3F). The initialization renders the change out of scope for the original map. I'll give it another go in a few hours following your recommendations, thanks for the pointers!